codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsJsonRpc1_0.java

```diff
@@ -36,4 +45,77 @@ protected String getDocumentContentType() {
     public ShapeId getProtocol() {
         return AwsJson1_0Trait.ID;
     }
+
+    @Override
+    protected Set<StructureShape> generateErrorShapes(
+        GenerationContext context, OperationShape operation, Symbol responseType) {
+        if (isAwsQueryCompatibleTraitFound(context)) {
+            return HttpProtocolGeneratorUtils.generateErrorDispatcher(
+                context, operation, responseType, this::writeErrorMessageCodeDeserializer,
+                this::getOperationErrors, (writer) -> AwsJsonRpc1_0.awsQueryCompatibleDefaultBlockWriter(writer));
+        } else {
+            return HttpProtocolGeneratorUtils.generateErrorDispatcher(
+                context, operation, responseType, this::writeErrorMessageCodeDeserializer,
+                this::getOperationErrors);
+        }
+    }
+
+    @Override
+    protected void deserializeError(GenerationContext context, StructureShape shape) {
```

Comment:
is aws json RPC 1.1 deprecated? reason why im asking is because if any other team (besides SQS) is migrating from AWS query to aws json. would they most likely be using aws json rpc 1.0?

Reply:
AWS Query Compatible is only compatible with the AWS JSON 1.0 protocol as per the smithy spec: https://smithy.io/2.0/aws/protocols/aws-query-protocol.html#aws-protocols-awsquerycompatible-trait.